### DISCUSSION TITLE
Take screenshot and get body HTML in once?

### DISCUSSION THREAD

**Discussion by Kian987** [spatie/browsershot#727] [Q&A] [answered]
Hi,

Is it possibile to use `->save()` and `->bodyHtml()` together? In essence taking a screenshot and retreive HTML source code in the same time without having to run two separate instances of Browsershot / chromium-browser.

I tried to combine them in different orders but I get errors, nothing or just one of the two working.

Any idea? Thanks

**Comment by Kian987** [comment, marked as answer]
Damn it. Right now I realized it was very simple.

```
// Initialize Browsershot with all your options
$instance = Browsershot::url($data['target']);

// Save image
$instance->save($path_to_image);

// Get DOM
$html = $instance->bodyHTML();
```

**Reply by klkvsk** [reply]
That would work, but will trigger two separate instances of puppeteer, and do two requests. 
Try to fetch something from random.org and see.
It's still unclear to me, how to send multiple commands to one instance.
